nesta/core/routines/arxiv/magrxiv_collect_iterative_task.py

```diff
@@ -0,0 +1,120 @@
+from nesta.core.luigihacks import misctools
+from nesta.core.luigihacks.mysqldb import MySqlTarget
+
+
+class CollectMagrxivTask(luigi.Task):
+    '''Collect bio/medrxiv articles from MAG and dump the
+    data in the MySQL server under the arxiv table.
+
+    Args:
+        date (datetime): Datetime used to label the outputs
+        db_config_env (str): environmental variable pointing to the db config file
+        db_config_path (str): The output database configuration
+        insert_batch_size (int): Number of records to insert into the database at once
+        articles_from_date (str): Earliest possible date considered to collect articles.
+    '''
+    date = luigi.DateParameter()
+    routine_id = luigi.Parameter()
+    test = luigi.BoolParameter(default=True)
+    xivs = luigi.ListParameter(default=['medrxiv', 'biorxiv'])
+    db_config_env = luigi.Parameter()
+    db_config_path = luigi.Parameter()
+    insert_batch_size = luigi.IntParameter(default=500)
+    articles_from_date = luigi.Parameter()
+
+    def output(self):
+        '''Points to the output database engine'''
+        db_config = misctools.get_config(self.db_config_path, "mysqldb")
+        db_config["database"] = 'dev' if self.test else 'production'
+        db_config["table"] = "arXlive <dummy>"  # Note, not a real table
+        update_id = "MagrxivCollect_{}".format(self.date)
+        return MySqlTarget(update_id=update_id, **db_config)
+
+    def run(self):
+        # database setup
+        database = 'dev' if self.test else 'production'
+        logging.warning(f"Using {database} database")
+        self.engine = get_mysql_engine(self.db_config_env, 'mysqldb', database)
+
+        mag_config = misctools.get_config('mag.config', "mag")
+        api_key = mag_config['subscription_key']
+
+        articles_from_date = pd.Timestamp(self.articles_from_date).to_pydatetime()
+        xiv_filter = Article.article_source.in_(self.xivs)
+        with db_session(self.engine) as session:
+            all_article_ids = {article.id for article in
+                               session.query(Article.id).filter(xiv_filter).all()}
+            logging.info(f"{len(all_article_ids)} existing articles")
+
+            # retrieve and process, while inserting any missing categories
+            articles = []
+            for xiv in self.xivs:
+                logging.info(f"Getting MAG data for '{xiv}'")
+                for row in get_magrxiv_articles(xiv, api_key, start_date=articles_from_date):
+                    if row['id'] in all_article_ids:
+                        continue
+                    all_article_ids.add(row['id'])
+                    articles.append(row)
+                    if not len(articles) % 1000:
+                        logging.info(f"Processed {len(articles)} articles")
+                        if self.test:
+                            break
+                if self.test and len(articles) > 1000:
```

Comment:
Is this meant to stop at `1000` or `2000`? I.e. should it be `>=`?
(Above line breaks *at* `1000`)

Reply:
ah, so actually the `>` ensures that it will do at least one more loop in test mode, to pick up multiple `xivs`. I'll add a comment to clarify that